mt_distributions/ExponentialDistribution.py

```diff
@@ -0,0 +1,206 @@
+        return "Lambda Parameter 1 {}, Lambda Parameter 2 {}".format(self.lp1, self.lp2)
+
+class Exponential(Distribution):
+    """ Exponential distribution Class for calculating and 
+	visualizing a Exponential distribution.
+	
+	Attributes:
+		lp (float) representing the decay parameter (λ) of the function
+		data_list (list of floats) a list of floats extracted from the data file
+	
+    """
+    def __init__(self, lp):
+        self.lp = lp
+
+        Distribution.__init__(self, self.calculate_mean(), self.calculate_stdev())
+    
+    def calculate_mean(self):
+        """Function to calculate the mean from lp
+        
+        Args: 
+            None
+        
+        Returns: 
+            float: mean of the distribution
+    
+        """
+        self.mean = 1/self.lp
+        return self.mean
+
+    def calculate_variance(self):
+        """Function to calculate the variance from lp
+        
+        Args: 
+            None
+        
+        Returns: 
+            float: variance of the distribution
+    
+        """
+        self.variance = 1/(self.lp*self.lp)
+        return self.variance
+    
+    def calculate_stdev(self):
+        """Function to calculate the standard deviation from the variance
+        
+        Args: 
+            None
+        
+        Returns: 
+            float: standard deviation of the distribution
+    
+        """
+        self.calculate_variance()
+        self.stdev = math.sqrt(self.variance)
+        return self.stdev
+    
+    def pdf(self, x):
+        """Probability density function calculator for the distribution.
+        
+        Args:
+            x (float): point for calculating the probability density function
+            
+        
+        Returns:
+            float: probability density function output
+        """
+        return self.lp * math.exp(-self.lp * x)
+    
+    def replace_stats_with_data(self):
+    
+        """Function to calculate p and n from the data set
+        
+        Args: 
+            None
+        
+        Returns: 
+            float: the p value
+            float: the n value
+    
+        """
+        n = len(self.data)
+        if n > 0:
+            mu = 1.0 * sum(self.data) / n
+            self.lp = 1/mu
+            self.calculate_mean()
+            self.calculate_stdev()
+        else:
+            raise IndexError('No data has been loaded')
+
+    def __add__(self, other):
+        """Function to add together two independent exponential distributions
+		
+		Args:
+			other (Exponential): ExponentialDistribution instance
+			
+		Returns:
+			An ErlangDistribution if both exponentials have the same lambda parameter
+            A HypoExponentialDistribution if both exponentials have differing lambda parameters
+			
+		"""
+        if isinstance(other, Exponential):
+            if other.lp == self.lp:
+                result = Erlang(self.lp*self.lp, 2)
+            else:
+                result = HypoExponential(self.lp, other.lp)
```

Comment:
It looks like here there is some dependency on HypoExponential. In this case, why not also include Erlang in the same file?

Reply:
I agree this wasn't consistent. I could either put the HE into it's own file, like the Erlang and import it or include the Erlang here. Let me know what you prefer as standard and I'll commit the changes.